scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+            method = self._logccdf_complementarity
+        else:
+            method = self._logccdf_quadrature
+        return method
+
+    def _logccdf_formula(self):
+        return NotImplementedError(self._not_implemented)
+
+    def _logccdf_logexp(self, x, **kwargs):
+        return np.log(self._ccdf_dispatch(x, **kwargs))
+
+    def _logccdf_complementarity(self, x, **kwargs):
+        return _log1mexp(self._logcdf_dispatch(x, **kwargs))
+
+    def _logccdf_quadrature(self, x, **kwargs):
+        _, b = self._support(**kwargs)
+        return self._quadrature(self._logpdf_dispatch, limits=(x, b),
+                                kwargs=kwargs, log=True)
+
+    def ccdf(self, x, y=None, *, method=None):
+        r"""Complementary cumulative distribution function
+
+        The complementary cumulative distribution function :math:`G(x)` is the
+        complement of the cumulative distribution function :math:`F(x)`; i.e.,
+        probability the random variable :math:`X` will take on a value greater
+        than :math:`x`:
+
+        .. math::
+
+            G(x) = 1 - F(x) = P(X > x)
+
+        A two-argument variant of this function is:
+
+        .. math::
+
+            G(x, y) = 1 - F(x, y) = P(X < x \text{ or } X > y)
+
+        ``ccdf`` accepts `x` for :math:`x` and `y` for :math:`y`.
+
+        Parameters
+        ----------
+        x, y : Array
+            The arguments of the complementary cumulative distribution
+            function (CCDF). `x` is required; `y` is optional.
+        method : {None, 'formula', 'logexp', 'complementarity', 'quadrature', 'addition'}
+            The strategy used to evaluate the CCDF.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the CCDF itself
+            - ``'logexp'``: evaluate the logarithm of the CCDF directly and
+                            exponentiate
+            - ``'complementarity'``: evaluate the CDF and take the complement
+            - ``'quadrature'``: numerically integrate the PDF
+
+            The two-argument form chooses between:
+
+            - ``'formula'``: use a formula for the CCDF itself
+            - ``'addition'``: compute the CDF at `x` and the CCDF at `y`, then add
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The CCDF evaluated at the provided argument(s).
+
+        See Also
+        --------
+        ContinuousDistribution.cdf
+        ContinuousDistribution.logccdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`.
+        The CCDF :math:`G(x)` is related to the probability density function
+        :math:`f(x)` by:
+
+        .. math::
+
+            G(a) = \int_a^r f(x) dx
```

Comment:
avoid a and b (see comment on cdf)

Reply:
Merging with https://github.com/mdhaber/scipy/pull/110#discussion_r1556426806